File: dyd/roots/libraries/dhall-codegen/dyd/assets/tests/example-stripe/go/out.go
package models


type StripeAddress struct {
	City *string `json:"city,omitempty"`
	Country *string `json:"country,omitempty"`
	Line1 *string `json:"line1,omitempty"`
	Line2 *string `json:"line2,omitempty"`
	PostalCode *string `json:"postal_code,omitempty"`
	State *string `json:"state,omitempty"`
}

type StripeCustomer struct {
	Balance int `json:"balance"`
	Created int `json:"created"`
	Id string `json:"id"`
	Object string `json:"object"`
	Address *StripeAddress `json:"address,omitempty"`
	Currency *string `json:"currency,omitempty"`
	Description *string `json:"description,omitempty"`
	Email *string `json:"email,omitempty"`
	Name *string `json:"name,omitempty"`
}

type StripeChargeBillingDetails struct {
	Address *StripeAddress `json:"address,omitempty"`
	Email *string `json:"email,omitempty"`
	Name *string `json:"name,omitempty"`
	Phone *string `json:"phone,omitempty"`
}

type StripeChargeOutcomeValue struct {
	Type string `json:"type"`
	AdviceCode *string `json:"advice_code,omitempty"`
	NetworkAdviceCode *string `json:"network_advice_code,omitempty"`
	NetworkDeclineCode *string `json:"network_decline_code,omitempty"`
	NetworkStatus *string `json:"network_status,omitempty"`
	Reason *string `json:"reason,omitempty"`
	RiskLevel *string `json:"risk_level,omitempty"`
	RiskScore *int `json:"risk_score,omitempty"`
	Rule *string `json:"rule,omitempty"`
	SellerMessage *string `json:"seller_message,omitempty"`
}

type StripeChargeRefundsValueKind string

const (
	StripeChargeRefundsValueKindOption1 StripeChargeRefundsValueKind = "option1"
	StripeChargeRefundsValueKindOption2 StripeChargeRefundsValueKind = "option2"
)

type StripeChargeRefundsValue struct {
	Kind StripeChargeRefundsValueKind `json:"kind"`
	Option1 *[]string `json:"option1,omitempty"`
	Option2 *[]StripeRefund `json:"option2,omitempty"`
}

type StripeCharge struct {
	Amount int `json:"amount"`
	AmountCaptured int `json:"amount_captured"`
	AmountRefunded int `json:"amount_refunded"`
	BillingDetails StripeChargeBillingDetails `json:"billing_details"`
	Captured bool `json:"captured"`
	Created int `json:"created"`
	Id string `json:"id"`
	Object string `json:"object"`
	Currency *string `json:"currency,omitempty"`
	Outcome *StripeChargeOutcomeValue `json:"outcome,omitempty"`
	Refunds *StripeChargeRefundsValue `json:"refunds,omitempty"`
}

type StripeRefundChargeKind string

const (
	StripeRefundChargeKindOption1 StripeRefundChargeKind = "option1"
	StripeRefundChargeKindOption2 StripeRefundChargeKind = "option2"
)

type StripeRefundCharge struct {
	Kind StripeRefundChargeKind `json:"kind"`
	Option1 *string `json:"option1,omitempty"`
	Option2 *StripeCharge `json:"option2,omitempty"`
}

type StripeRefund struct {
	Amount int `json:"amount"`
	Charge StripeRefundCharge `json:"charge"`
	Id string `json:"id"`
	Object string `json:"object"`
}

File: dyd/roots/libraries/dhall-codegen/dyd/assets/tests/example-stripe/py-3.8/out.py

from datetime import datetime
from pydantic import BaseModel, Field, TypeAdapter
from typing import (
    Any, Optional, Union, List, Set, Dict, Tuple, Callable, Literal
)

# backports for 3.8:
from typing_extensions import Annotated, Protocol  # for constraints/validators

try:
    from typing_extensions import TypeAlias
except Exception:
    TypeAlias = Any  # fall back; plain assignment still works

try:
    from types import NoneType          # 3.10+
except ImportError:
    NoneType = type(None)               # fallback for 3.8–3.9

# used to simulate intersection type
def _check_all_of(*adapters: TypeAdapter[Any]):
    def check(v: Any) -> Any:
        for ta in adapters:
            ta.validate_python(v)  # raises on failure
        return v
    return check




class StripeAddress(BaseModel):
    city : Optional[str]
    country : Optional[str]
    line1 : Optional[str]
    line2 : Optional[str]
    postal_code : Optional[str]
    state : Optional[str]


class StripeCustomer(BaseModel):
    balance : int
    created : int
    id : str
    object : str
    address : Optional[StripeAddress]
    currency : Optional[str]
    description : Optional[str]
    email : Optional[str]
    name : Optional[str]


class StripeChargeBillingDetails(BaseModel):
    address : Optional[StripeAddress]
    email : Optional[str]
    name : Optional[str]
    phone : Optional[str]


class StripeChargeOutcomeValue(BaseModel):
    type : str
    advice_code : Optional[str]
    network_advice_code : Optional[str]
    network_decline_code : Optional[str]
    network_status : Optional[str]
    reason : Optional[str]
    risk_level : Optional[str]
    risk_score : Optional[int]
    rule : Optional[str]
    seller_message : Optional[str]


class StripeCharge(BaseModel):
    amount : int
    amount_captured : int
    amount_refunded : int
    billing_details : StripeChargeBillingDetails
    captured : bool
    created : int
    id : str
    object : str
    currency : Optional[str]
    outcome : Optional[StripeChargeOutcomeValue]
    refunds : Optional[Union[List[str],List[StripeRefund]]]


class StripeRefund(BaseModel):
    amount : int
    charge : Union[str,StripeCharge]
    id : str
    object : str

